eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/microfragments/eventdetails/EventDescriptionText.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.microfragments.eventdetails;
+
+import com.schedjoules.eventdiscovery.framework.utils.charsequence.Trimmed;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+import com.schedjoules.eventdiscovery.framework.utils.spanned.Html;
+import com.schedjoules.eventdiscovery.framework.utils.spanned.Linkified;
+
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+
+
+/**
+ * Represents the formatted event description text.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class EventDescriptionText extends AbstractCachingOptional<CharSequence>
+{
+    public EventDescriptionText(final Optional<String> description)
+    {
+        super(new Factory<Optional<CharSequence>>()
+        {
+            @Override
+            public Optional<CharSequence> create()
+            {
+                if (!description.isPresent())
+                {
+                    return Absent.absent();
+                }
+
+                String desc = description.value();
+                CharSequence formatted = isHtml(desc) ? new Html(desc) : new Linkified(new Trimmed(desc));
```

Comment:
Absolutely right, thanks, I've updated it, looks better this way. I've also added `OptionalMapped` that maps from optional to optional - not sure if that's a good name.
Also, I've noticed an embarrassing bug in Trimmed, fixed that, added unit test.
Please check again.